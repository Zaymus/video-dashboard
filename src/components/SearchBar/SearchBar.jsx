import { useRef } from 'react';
import styled from "styled-components";
import { SROnly } from '../common';

const SearchContainer = styled.div`
  width: 35%;
  height: 40px;
  border-radius: 25px;
  border: 2px solid var(--primary);
  background: var(--surface-dark);
  display: flex;
  align-items: center;
  overflow: hidden;
`;

const SearchInput = styled.input`
  height: 100%;
  flex: 1;
  padding: 0 10px 0 20px;
  font-size: 1rem;
  border: none;
  background-color: var(--background-dark);
  color: var(--text-dark);

  &:focus {
    outline: none;
  } 
`;

const SearchButton = styled.button`
  height: 100%;
  flex: 0.1;
  border: none;
  font-size: 1.2rem;
  background-color: var(--surface-dark);
  color: var(--primary);
  transition: text-shadow 0.15s linear;
  border-radius: 0 25px 25px 0px;

  &:hover {
    cursor: pointer;
    text-shadow: 0px 0px 5px var(--primary);
  }
`;

const SearchBar = () => {
  const searchBarRef = useRef(null);

  const focusHandler = () => {
    if(searchBarRef) {
      searchBarRef.current.style.transition = "box-shadow 0.15s linear";
      searchBarRef.current.style.boxShadow = "0px 0px 10px var(--primary)";
    }
  }

  const blurHandler = () => {
    if (searchBarRef) {
      searchBarRef.current.style.boxShadow = null;
    }
  }

  return (
    <SearchContainer ref={searchBarRef}>
      <SearchInput
        type="search"
        placeholder="Search"
        name="search"
        onFocus={focusHandler}
        onBlur={blurHandler}
        spellCheck
      />
      <SearchButton>
        <SROnly>Search</SROnly>
        <i className="fa-solid fa-magnifying-glass"></i>
      </SearchButton>
    </SearchContainer>
  );
}

export default SearchBar;