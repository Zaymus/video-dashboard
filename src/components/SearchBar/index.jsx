import { lazy } from 'react';

const SearchBar = () => lazy(() => import('./SearchBar.jsx'));

export default SearchBar;