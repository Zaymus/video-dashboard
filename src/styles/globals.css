:root {
  --primary: #1E90FF;
  --secondary: #FF6B6B;
  --background-dark: #121212;
  --surface-dark: #1F1F1F;
  --text-primary-dark: #FFFFFF;
  --text-secondary-dark: #B3B3B3;
  --border-dark: #2C2C2C;
  --success: #2ECC71;
  --error: #E74C3C;
  --warning: #F1C40F;
  --background-light: #F9F9F9;
  --surface-light: #FFFFFF;
  --text-primary-light: #1A1A1A;
  --text-secondary-light: #4F4F4F;
  --border-light: #E0E0E0;

  --videos-gutter-margin: 30px;
  --videos-per-row: 5;
  --videos-usable-width: calc(100% - var(--videos-gutter-margin) * 2);
}

body {
  background-color: var(--background-dark);
  margin: 0;
  color: var(--text-primary-dark);
  font-size: 16pt;
}

#root {
  min-height: 100vh;
}