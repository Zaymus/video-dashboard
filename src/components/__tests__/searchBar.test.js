import { jest, describe, test, expect, beforeAll, afterAll } from '@jest/globals';
import { fireEvent, render, screen, act } from '@testing-library/react';
import React from 'react';

import SearchBar from '../SearchBar/SearchBar';

describe('SearchBar component testing', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  test('handlers work as intended with no ref', () => {
    jest.spyOn(React, 'useRef').mockReturnValueOnce(null);
    const { container } = render(<SearchBar />);

    fireEvent["focus"](screen.getByPlaceholderText("Search"));
    expect(container).toMatchSnapshot();
    fireEvent["blur"](screen.getByPlaceholderText("Search"));
    expect(container).toMatchSnapshot();
  });
  test('matches snapshot', () => {
    const { container } = render(<SearchBar />);
    expect(container).toMatchSnapshot();
  });

  test('matches snapshot - focus', () => {
    const { container } = render(<SearchBar />);
    fireEvent["focus"](screen.getByPlaceholderText("Search"));
    act(() => {
      jest.advanceTimersByTime(500);
    });
    expect(container).toMatchSnapshot();
  });

  test('matches snapshot - blur', () => {
    const { container } = render(<SearchBar />);
    fireEvent['focus'](screen.getByPlaceholderText("Search"));
    act(() => {
      jest.advanceTimersByTime(500);
    });

    fireEvent["blur"](screen.getByPlaceholderText("Search"));
    act(() => {
      jest.advanceTimersByTime(500);
    });
    expect(container).toMatchSnapshot();
  });

  test('matches snapshot - button hover', () => {
    const { container } = render(<SearchBar />);
    fireEvent["mouseEnter"](screen.getByPlaceholderText("Search"));
    act(() => {
      jest.advanceTimersByTime(500);
    });
    expect(container).toMatchSnapshot();
  });

  test('matches snapshot - button blur', () => {
    const { container } = render(<SearchBar />);
    fireEvent["mouseEnter"](screen.getByPlaceholderText("Search"));
    act(() => {
      jest.advanceTimersByTime(500);
    });

    fireEvent["mouseLeave"](screen.getByPlaceholderText("Search"));
    act(() => {
      jest.advanceTimersByTime(500);
    });
    expect(container).toMatchSnapshot();
  });
});