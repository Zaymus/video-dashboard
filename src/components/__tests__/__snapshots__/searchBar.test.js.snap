// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SearchBar component testing handlers work as intended with no ref 1`] = `
<div>
  <div
    class="sc-ggWZvA ZMjlW"
  >
    <input
      class="sc-dTvVRJ FRDnM"
      name="search"
      placeholder="Search"
      spellcheck="true"
      type="search"
    />
    <button
      class="sc-jwTyAe eKrelh"
    >
      <span
        class="sc-fhHczv jxaKYX"
      >
        Search
      </span>
      <i
        class="fa-solid fa-magnifying-glass"
      />
    </button>
  </div>
</div>
`;

exports[`SearchBar component testing handlers work as intended with no ref 2`] = `
<div>
  <div
    class="sc-ggWZvA ZMjlW"
  >
    <input
      class="sc-dTvVRJ FRDnM"
      name="search"
      placeholder="Search"
      spellcheck="true"
      type="search"
    />
    <button
      class="sc-jwTyAe eKrelh"
    >
      <span
        class="sc-fhHczv jxaKYX"
      >
        Search
      </span>
      <i
        class="fa-solid fa-magnifying-glass"
      />
    </button>
  </div>
</div>
`;

exports[`SearchBar component testing matches snapshot - blur 1`] = `
<div>
  <div
    class="sc-ggWZvA ZMjlW"
    style="transition: box-shadow 0.15s linear;"
  >
    <input
      class="sc-dTvVRJ FRDnM"
      name="search"
      placeholder="Search"
      spellcheck="true"
      type="search"
    />
    <button
      class="sc-jwTyAe eKrelh"
    >
      <span
        class="sc-fhHczv jxaKYX"
      >
        Search
      </span>
      <i
        class="fa-solid fa-magnifying-glass"
      />
    </button>
  </div>
</div>
`;

exports[`SearchBar component testing matches snapshot - button blur 1`] = `
<div>
  <div
    class="sc-ggWZvA ZMjlW"
  >
    <input
      class="sc-dTvVRJ FRDnM"
      name="search"
      placeholder="Search"
      spellcheck="true"
      type="search"
    />
    <button
      class="sc-jwTyAe eKrelh"
    >
      <span
        class="sc-fhHczv jxaKYX"
      >
        Search
      </span>
      <i
        class="fa-solid fa-magnifying-glass"
      />
    </button>
  </div>
</div>
`;

exports[`SearchBar component testing matches snapshot - button hover 1`] = `
<div>
  <div
    class="sc-ggWZvA ZMjlW"
  >
    <input
      class="sc-dTvVRJ FRDnM"
      name="search"
      placeholder="Search"
      spellcheck="true"
      type="search"
    />
    <button
      class="sc-jwTyAe eKrelh"
    >
      <span
        class="sc-fhHczv jxaKYX"
      >
        Search
      </span>
      <i
        class="fa-solid fa-magnifying-glass"
      />
    </button>
  </div>
</div>
`;

exports[`SearchBar component testing matches snapshot - focus 1`] = `
<div>
  <div
    class="sc-ggWZvA ZMjlW"
    style="transition: box-shadow 0.15s linear; box-shadow: 0px 0px 10px var(--primary);"
  >
    <input
      class="sc-dTvVRJ FRDnM"
      name="search"
      placeholder="Search"
      spellcheck="true"
      type="search"
    />
    <button
      class="sc-jwTyAe eKrelh"
    >
      <span
        class="sc-fhHczv jxaKYX"
      >
        Search
      </span>
      <i
        class="fa-solid fa-magnifying-glass"
      />
    </button>
  </div>
</div>
`;

exports[`SearchBar component testing matches snapshot 1`] = `
<div>
  <div
    class="sc-ggWZvA ZMjlW"
  >
    <input
      class="sc-dTvVRJ FRDnM"
      name="search"
      placeholder="Search"
      spellcheck="true"
      type="search"
    />
    <button
      class="sc-jwTyAe eKrelh"
    >
      <span
        class="sc-fhHczv jxaKYX"
      >
        Search
      </span>
      <i
        class="fa-solid fa-magnifying-glass"
      />
    </button>
  </div>
</div>
`;
